test/hooks/permissionedPools/PermissionedPositionManager.t.sol

```diff
@@ -0,0 +1,522 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.26;
+
+import "forge-std/Test.sol";
+import {PoolManager} from "@uniswap/v4-core/src/PoolManager.sol";
+import {IPoolManager} from "@uniswap/v4-core/src/interfaces/IPoolManager.sol";
+import {IHooks} from "@uniswap/v4-core/src/interfaces/IHooks.sol";
+import {Currency} from "@uniswap/v4-core/src/types/Currency.sol";
+import {PoolId} from "@uniswap/v4-core/src/types/PoolId.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
+import {BalanceDelta} from "@uniswap/v4-core/src/types/BalanceDelta.sol";
+import {LiquidityAmounts} from "@uniswap/v4-core/test/utils/LiquidityAmounts.sol";
+import {TickMath} from "@uniswap/v4-core/src/libraries/TickMath.sol";
+import {FixedPointMathLib} from "solmate/src/utils/FixedPointMathLib.sol";
+import {StateLibrary} from "@uniswap/v4-core/src/libraries/StateLibrary.sol";
+import {LPFeeLibrary} from "@uniswap/v4-core/src/libraries/LPFeeLibrary.sol";
+import {IAllowanceTransfer} from "permit2/src/interfaces/IAllowanceTransfer.sol";
+import {Position} from "@uniswap/v4-core/src/libraries/Position.sol";
+import {SafeCast} from "@uniswap/v4-core/src/libraries/SafeCast.sol";
+import {CREATE3} from "solmate/src/utils/CREATE3.sol";
+
+import {MockERC20} from "solmate/src/test/utils/mocks/MockERC20.sol";
+import {IERC721} from "forge-std/interfaces/IERC721.sol";
+
+import {IPositionManager} from "../../../src/interfaces/IPositionManager.sol";
+import {Actions} from "../../../src/libraries/Actions.sol";
+import {DeltaResolver} from "../../../src/base/DeltaResolver.sol";
+import {PositionConfig} from "test/shared/PositionConfig.sol";
+import {SlippageCheck} from "../../../src/libraries/SlippageCheck.sol";
+import {BaseActionsRouter} from "../../../src/base/BaseActionsRouter.sol";
+import {ActionConstants} from "../../../src/libraries/ActionConstants.sol";
+
+import {LiquidityFuzzers} from "../../shared/fuzz/LiquidityFuzzers.sol";
+import {Planner, Plan} from "../../shared/Planner.sol";
+import {PermissionedPosmTestSetup} from "./shared/PermissionedPosmTestSetup.sol";
+import {ReentrantToken} from "../../mocks/ReentrantToken.sol";
+import {ReentrancyLock} from "../../../src/base/ReentrancyLock.sol";
+import {ModifyLiquidityParams} from "@uniswap/v4-core/src/types/PoolOperation.sol";
+
+import {MockAllowlistChecker, MockPermissionedToken} from "./PermissionedPoolsBase.sol";
+import {WrappedPermissionedToken, IERC20} from "../../../src/hooks/permissionedPools/WrappedPermissionedToken.sol";
+import {WrappedPermissionedTokenFactory} from "../../../src/hooks/permissionedPools/WrappedPermissionedTokenFactory.sol";
+import {Hooks} from "@uniswap/v4-core/src/libraries/Hooks.sol";
+import {SortTokens} from "@uniswap/v4-core/test/utils/SortTokens.sol";
+
+contract PermissionedPositionManagerTest is Test, PermissionedPosmTestSetup, LiquidityFuzzers {
```

Comment:
There should be tests covering the 4 branches in the overridden _pay function:
1. non permissioned token - caller is payer
2. non permissioned token - posm is payer
3. permissioned token - caller is payer
4. permissioned token - posm is payer
same for the v4 router